core/src/test/java/org/apache/iceberg/jdbc/TestJdbcUtil.java

```diff
@@ -45,4 +51,114 @@ public void testFilterAndRemovePrefix() {
 
     assertThat(expected).isEqualTo(actual);
   }
+
+  @Test
+  public void testV0toV1SqlStatements() throws Exception {
```

Comment:
I think it would make sense to have such a test where the Catalog API is used to create a table/view rathen than the underlying SQL statements

Reply:
I can do that in the `TestJdbcCatalog` using the API. The reason of this test was to focus on SQL statements.

Let me move this test in `TestJdbcCatalog`.

NB: I will check if the `TestTableCatalog` covers the commit workflow/use case.

Reply:
it's probably ok to keep these tests here, but users will always interact via the Catalog API, so we need to make sure that stuff works when such an upgrade is done and the Catalog API is used

Reply:
Agree, let me add a new test in `TestJdbcCatalog` using the facing API. Thanks @nastra !